label_studio/users/views.py

```diff
@@ -77,13 +78,19 @@ def user_signup(request):
     """
     user = request.user
     next_page = request.GET.get('next')
+    token = request.GET.get('token')
     next_page = next_page if next_page else reverse('projects:project-index')
     user_form = forms.UserSignupForm()
     organization_form = OrganizationSignupForm()
 
     if user.is_authenticated:
         return redirect(next_page)
 
+    organization = Organization.objects.first()
+    if settings.DISABLE_SIGNUP_WITHOUT_LINK is True:
+        if not(token and organization and token == organization.token):
```

Comment:
I think this is wrong. We should check token on auth (POST), not GET

Reply:
Moved check inside POST processing and fixed tests.